sycl/include/CL/sycl/id.hpp

```diff
@@ -119,22 +120,119 @@ template <int dimensions = 1> class id : public detail::array<dimensions> {
 
-  #undef __SYCL_GEN_OPT
+#undef __SYCL_GEN_OPT
 
-  // OP is: +=, -=, *=, /=, %=, <<=, >>=, &=, |=, ^=
-  #define __SYCL_GEN_OPT(op)                                                   \
-    id<dimensions> &operator op(const id<dimensions> &rhs) {                   \
-      for (int i = 0; i < dimensions; ++i) {                                   \
-        this->common_array[i] op rhs.common_array[i];                          \
-      }                                                                        \
-      return *this;                                                            \
+// OP is: +=, -=, *=, /=, %=, <<=, >>=, &=, |=, ^=
+#define __SYCL_GEN_OPT(op)                                                     \
+  id<dimensions> &operator op(const id<dimensions> &rhs) {                     \
+    for (int i = 0; i < dimensions; ++i) {                                     \
+      this->common_array[i] op rhs.common_array[i];                            \
     }                                                                          \
-    id<dimensions> &operator op(const size_t &rhs) {                           \
-      for (int i = 0; i < dimensions; ++i) {                                   \
-        this->common_array[i] op rhs;                                          \
-      }                                                                        \
-      return *this;                                                            \
+    return *this;                                                              \
+  }                                                                            \
+  id<dimensions> &operator op(const size_t &rhs) {                             \
+    for (int i = 0; i < dimensions; ++i) {                                     \
+      this->common_array[i] op rhs;                                            \
     }                                                                          \
+    return *this;                                                              \
+  }
+
+  __SYCL_GEN_OPT(+=)
+  __SYCL_GEN_OPT(-=)
+  __SYCL_GEN_OPT(*=)
+  __SYCL_GEN_OPT(/=)
+  __SYCL_GEN_OPT(%=)
+  __SYCL_GEN_OPT(<<=)
+  __SYCL_GEN_OPT(>>=)
+  __SYCL_GEN_OPT(&=)
+  __SYCL_GEN_OPT(|=)
+  __SYCL_GEN_OPT(^=)
+
+#undef __SYCL_GEN_OPT
+};
+
+template <> class id<1> : public detail::array<1> {
+private:
+  using base = detail::array<1>;
+
+public:
+  id() = default;
+
+  /* The following constructor is only available in the id struct
+   * specialization where: dimensions==1 */
+  id<1>(size_t dim0) : base(dim0) {}
+
+  id<1>(const range<1> &range_size) : base(range_size.get(0)) {}
+
+  template <bool with_offset = true>
+  id<1>(const item<1, with_offset> &item) : base(item.get_id(0)) {}
+
+  explicit operator range<1>() const {
+    range<1> result(detail::InitializedVal<1, range>::template get<0>());
+    result[0] = this->get(0);
+    return result;
+  }
+
+  operator size_t() const { return this->get(0); }
```

Comment:
Could you please guard this new non-standard method with some preprocessor macro? Similar to how it is done in #974 

It would be also good to have another one macro which allows to disable all non-standard features introduced by header files

Reply:
added "__SYCL_DISABLE_ID_TO_INT_CONV__"